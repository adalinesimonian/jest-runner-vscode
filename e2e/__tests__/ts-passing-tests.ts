import path from 'path'
import { runJest } from '../utils'

const testPath = path.resolve(__dirname, '../ts-passing-tests')

describe('Passing tests (ts-jest)', () => {
  it('should pass tests', async () => {
    const { exitCode, stderr, stdout, json } = await runJest(testPath)
    if (exitCode !== 0) {
      console.error('stderr', stderr)
      console.error('stdout', stdout)
    }
    expect(json).toMatchSnapshot()
    expect(exitCode).toBe(0)
  }, 30000)
})
