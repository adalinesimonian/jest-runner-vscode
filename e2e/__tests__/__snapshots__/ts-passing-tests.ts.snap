// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passing tests (ts-jest) should pass tests 1`] = `
{
  "numFailedTestSuites": 0,
  "numFailedTests": 0,
  "numPassedTestSuites": 1,
  "numPassedTests": 2,
  "numPendingTestSuites": 0,
  "numPendingTests": 0,
  "numRuntimeErrorTestSuites": 0,
  "numTodoTests": 0,
  "numTotalTestSuites": 1,
  "numTotalTests": 2,
  "openHandles": [],
  "snapshot": {
    "added": 0,
    "didUpdate": false,
    "failure": false,
    "filesAdded": 0,
    "filesRemoved": 0,
    "filesRemovedList": [],
    "filesUnmatched": 0,
    "filesUpdated": 0,
    "matched": 0,
    "total": 0,
    "unchecked": 0,
    "uncheckedKeysByFile": [],
    "unmatched": 0,
    "updated": 0,
  },
  "startTime": undefined,
  "success": true,
  "testResults": [
    {
      "assertionResults": [
        {
          "ancestorTitles": [
            "Describe",
          ],
          "duration": undefined,
          "failureMessages": [],
          "fullName": "Describe should test",
          "location": null,
          "status": "passed",
          "title": "should test",
        },
        {
          "ancestorTitles": [
            "Describe",
          ],
          "duration": undefined,
          "failureMessages": [],
          "fullName": "Describe should test async",
          "location": null,
          "status": "passed",
          "title": "should test async",
        },
      ],
      "endTime": undefined,
      "failureMessage": "",
      "message": "",
      "name": "/e2e/ts-passing-tests/__tests__/test.ts",
      "startTime": undefined,
      "status": "passed",
      "summary": "",
    },
  ],
  "wasInterrupted": false,
}
`;
